addons/website_forum/controllers/website_forum.py

```diff
@@ -179,31 +180,53 @@ def tag_read(self, forum_id, query='', limit=25, **post):
             headers=[("Content-Type", "application/json")]
         )
 
-    @http.route(['/forum/<model("forum.forum"):forum>/tag', '/forum/<model("forum.forum"):forum>/tag/<string:tag_char>'], type='http', auth="public", website=True, sitemap=False)
-    def tags(self, forum, tag_char=None, **post):
-        # build the list of tag first char, with their value as tag_char param Ex : [('All', 'all'), ('C', 'c'), ('G', 'g'), ('Z', z)]
-        first_char_tag = forum.get_tags_first_char()
-        first_char_list = [(t, t.lower()) for t in first_char_tag if t.isalnum()]
-        first_char_list.insert(0, (_('All'), 'all'))
+    @http.route(['/forum/<model("forum.forum"):forum>/tag',
+                 '/forum/<model("forum.forum"):forum>/tag/<string:tag_char>',
+                 ], type='http', auth="public", website=True, sitemap=False)
+    def tags(self, forum, tag_char=None, filters='all', search='', **post):
+        # Build tags result without using tag_char to build pager, then return tags matching it
+        values = self._prepare_user_values(forum=forum, searches={'tags': True}, **post)
+        request.env["forum.post"].flush_model()
+        tags = request.env["forum.tag"]
+        order = 'posts_count DESC' if tag_char and tag_char != "all" else 'name'
 
-        active_char_tag = tag_char and tag_char.lower() or 'all'
+        if search:
+            values.update(search=search)
+            __, details, __ = request.website._search_with_fuzzy(
+                'forum_tags_only', search, limit=None, order=order, options={'forum': forum},
+            )
+            tags = details[0].get('results', tags)
+
+        if filters in ('unused', 'most_used'):
+            filter_tags = forum.tag_most_used_ids if filters == 'most_used' else forum.tag_unused_ids
+            tags = tags & filter_tags if tags else filter_tags
+
+        elif filters in ('all', 'followed'):
+            domain = [('forum_id', '=', forum.id), ('posts_count', '>', 0)]
+            if filters == 'followed' and not request.env.user._is_public():
+                domain = expression.AND([domain, [('message_is_follower', '=', True)]])
+            if search:
+                tags = tags.filtered_domain(domain) if tags else tags
```

Comment:
can't the domain be included in `_search_with_fuzzy` directly ? (with `option` and `base_domain`)

Reply:
We would have to duplicate the domain elements to search with them only for `all` and `followed` filters, while filtered_domain is not very costly. It's true that search_with_fuzzy would return unnecessary results, but we are unlikely to have hundreds of tags too...

Reply:
"but we are unlikely to have hundreds of tags too" ... not sure for that :D